src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/MethodTypeInference.cs

```diff
@@ -556,7 +556,7 @@ private void InferTypeArgsFirstPhase(Binder binder, ref HashSet<DiagnosticInfo>
             {
                 BoundExpression argument = _arguments[arg];
                 TypeWithAnnotations target = _formalParameterTypes[arg];
-                ExactOrBoundsKind kind = GetRefKind(arg).IsManagedReference() || target.Type.IsPointerType() ? ExactOrBoundsKind.Exact : ExactOrBoundsKind.LowerBound;
+                ExactOrBoundsKind kind = GetRefKind(arg).IsManagedReference() || target.Type.IsPointerOrFunctionPointer() ? ExactOrBoundsKind.Exact : ExactOrBoundsKind.LowerBound;
```

Comment:
>IsPointerOrFunctionPointer [](start = 93, length = 26)

I would expect that we would support inference somewhat similar to delegates

Reply:
Added a note to [https://github.com/dotnet/roslyn/issues/39865](https://github.com/dotnet/roslyn/issues/39865).

---
In reply to: [406388004](https://github.com/dotnet/roslyn/pull/43041#discussion_r406388004) [](ancestors = 406388004)